cmd/crossplane/stack/manage/manage.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manage
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/alecthomas/kingpin.v2"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane/apis"
+	"github.com/crossplane/crossplane/pkg/controller/stacks"
+	"github.com/crossplane/crossplane/pkg/controller/stacks/templates"
+)
+
+// Command configuration for the stack manager.
+type Command struct {
+	Name                      string
+	Sync                      time.Duration
+	RestrictCoreAPIGroups     bool
+	EnableTemplateStacks      bool
+	TemplatingControllerImage string
+	HostControllerNamespace   string
+	TenantKubeConfig          string
+}
+
+// FromKingpin produces the stack manager command from a Kingpin command.
+func FromKingpin(cmd *kingpin.CmdClause) Command {
+	c := Command{Name: cmd.FullCommand()}
+	cmd.Flag("sync", "Controller manager sync period duration such as 300ms, 1.5h or 2h45m").Short('s').Default("1h").DurationVar(&c.Sync)
+	cmd.Flag("restrict-core-apigroups", "Enable API group restrictions for Stacks. When enabled, APIs that Stacks depend on and own must contain a dot (\".\") and may not end with \"k8s.io\". When omitted, all groups are permitted.").Default("false").BoolVar(&c.RestrictCoreAPIGroups)
+	cmd.Flag("templates", "Enable support for template stacks").BoolVar(&c.EnableTemplateStacks)
+	cmd.Flag("templating-controller-image", "The image of the template stacks controller").StringVar(&c.TemplatingControllerImage)
```

Comment:
Shameful admission: I changed the behaviour from automatically enabling template stacks when a controller image was specified to just ignoring the templating-controller-image if template stacks isn't enabled because this function was over our complexity goal with all the error checks. If we really want that behaviour we could also just `//nolint` around the cyclomatic complexity check.